### DISCUSSION TITLE
Animated Image Support (badges/icon/profile pics)

### DISCUSSION THREAD

**Discussion by ghost** [RetroAchievements/RAWeb#766] [Ideas]
It would be cool to have Animated Badge support on the Game Page which can help the gamer to identify what the purpose/goal of the Cheevo is similar to a short Guide as it can display more information as a regular Badge due to shared information across multiple frames.

Additional to this, a mouse rollover function for the Game Page to play the Animated Badge even when the Cheevo is grayed out (hasn't been earned yet).

Examples:
![Animated Badge 1](https://i.imgur.com/TkQaEgE.gif)
![Animated Badge 1](https://i.imgur.com/t4Epp65.gif)
![Animated Badge 1](https://i.imgur.com/ohhRdKZ.gif)
![Animated Badge 1](https://i.imgur.com/vtN8jNf.gif)

**Comment by Jamiras** [comment]
I am not fond of this idea. Any form of motion attracts attention and will distract the user from the rest of the content on the page. 

If the description is not sufficient for determining the "purpose/goal" of the achievement, additional detail (and/or a video link illustrating the achievement) should be provided in the achievement details page.

Besides, do you really want your profile page to look like this? https://youtu.be/6WpMlwVwydo?t=8

**Comment by ScottFromDerby** [comment]
I just want to quickly weigh in - I love this idea! I completely don't want a hamster-dance profile page, but I really do think there's a place for animated badges/profile icons etc. I think we should be able to weigh up when they are appropriate, and good places to include animations like this. Definitely needs further thought and discussion, but I'm really interested in what we can do with this. We should also consider what happens in the client too with the pop-up - should that animate too?

**Comment by ghost** [comment]
@Jamiras: I don't see the reference to Hamster Dance to be honest, first the Hamster video has audio, animated gifs don't, also the frames in animated gifs aren't as agressive as a video is. If anything then there should be a setting to disable epileptic content, for example disabling Animated Badges (using the static image version of it) this setting could be present in each users Profile, just like the display of mature content. Some people want it, some don't.

@ScottFromDerby: IMO it would be cool if the emulators support this as well, we aren't living in the 90s anymore so i don't see why such relative simply thing can't be done with modern technology. We also be the first who do this stuff, not XBL, PSN or Steam has it, this is the benefits of not working with restrictions.

I see a lot of positive opportunities in Animated Badges, we can share much more information and motion based hints in badges to give the gamers a much better understanding what they suppose to do. Its easier to understand a motion-picure than just a non-motion-picture.

These examples posted above are just super simply versions of what is possible. Its just to give you the idea how it could look.

**Comment by ScottFromDerby** [comment]
sounds cool, just to follow up, agreed that a modern technology advances does mean that developing a program that contains an animated image is a fairly simple thing to achieve as a programmer. However, animating a dynamically loaded image in a series of varying tiers of emulators and rendering strategies is something that, while a programmer with spare time could eventually achieve it, is something that we may be missing the coder time/ability for right now. Not that it can't happen eventually, but we should consider some sort of fallback as well.

**Comment by ScottFromDerby** [comment]
As you mention, while modern technology does allow for these ideas, be aware that Microsoft and Sony with all the capability in the world have not yet managed to produce this into their ecosystems; this should be just as much an indicator for what is doable

**Comment by Thoooreau** [comment]
Looks great on themselves, but I can see the cons of a "motion fest", too. What if it animates only on mouse hover? Let's say on game page: it could have a very small icon like a sparkle ✨ next to the badges/icons that has an animation, on top left, let's say.

**Comment by Jamiras** [comment]
>  I don't see the reference to Hamster Dance

The profile page is 90% badge images. If they were all animated, it would just be a huge block of randomly gyrating images.

> We also be the first who do this stuff

There might be a reason for that. I doubt you're the first to suggest the idea. There's probably been a case study. 

In general, I can't think of a single user interface where non-subtle animations occur without user interaction (i.e. on mouse over or click).

> we aren't living in the 90s 

Maybe not, but the toolkit is written using APIs from the 90s. All rendering is done using raw Windows APIs (BitBlt, etc). We're not even using GDI+, let alone DirectX or SDL/Allegro/Unity.

**Comment by ghost** [comment]
Regards the technology limitations, with RetroArch and its new Widget System i don't see why it shouldn't be possible across many platformers, the Widget System works even on Systems like PS3 etc. from what i read there are also more plans for the Widget System in the future so this could be one of it. IMO there is no point to rely on those Standalone emulators anymore beside PJ64 anyway. As for the Game Page, it could be turned on/off easily in the settings for those who don't like it. Why enforcing something on everyone globally when some people want/don't want it.

@Jamiras: We can't buy anything with ideas, but we can share them to make changes. i am not getting anything from sharing ideas. Its your decision to make the call, you are the Developer, i am just a Content Creator who supports this idea.

**Comment by kdecks** [comment]
This isn't a new concept. Is there an open issue with it? (Looked but couldn't find a related one) I mean it's been discussed a handful of time in different places, also animated profile images, animate game icons.

Having the option, having the option be OPT-IN and not hiding the option in some deep corner that no one would ever see (Like if a game had animated icons include a checkmark on the game page.), that's how I'd approach it. That or or thoreau suggested it.

It has to be opt-in (or mouse over), because hampsterdance would would be an eyesore at least for some people.

Options and customization are always cool!

**Comment by natinusala** [comment]
RetroArch can be extended to add GIF capabilities, at least to the notifications. Not sure about the menu, it may look like an animated mess with all the GIFs playing at the same time. However we can animate the currently selected achievement, that would look good.

**Comment by ghost** [comment]
@natinusala: I agree, playing them all at once would be to much. Only the selected would be perfectly fine.

**Comment by andres-asm** [comment]
I think optinal animated GIFs or PNGs would be nice.
For the site you can have it just play them back on hover, and that's pretty much trivial to implement.

For the clients it's probably a bit more complex.

**Comment by kdecks** [comment]
> Why is everyone so obsessed with owning ideas? fill a patent then it belongs to you

I never said it was MY idea, good ideas have a way of resurfacing from various people. The point is the concept has been bounced around plenty. That there's no issue for it means it's good here. I think it's alright as a concept. Would be most potent for game and users profiles.
